OptimoveSDK/Sources/Classes/Optimobile/Network/AuthorizationMediator.swift

```diff
@@ -0,0 +1,67 @@
+//  Copyright © 2023 Optimove. All rights reserved.
+
+import Foundation
+
+enum AuthorizationStrategy {
+    case basic
+    // case bearer
+}
+
+protocol AuthorizationMediatorProtocol {
+    func getAuthorization(_: AuthorizationStrategy) throws -> String
+}
+
+enum HttpAuthorizationError: Error {
+    case missingAuthHeader
+}
+
+protocol HttpAuthorizationProtocol {
+    func authorizeRequest(_: inout URLRequest) throws
+}
+
+typealias CredentialsProvider = () -> OptimobileCredentials?
+
+final class AuthorizationMediator {
+    var basicAuthorization: String?
+    let provider: CredentialsProvider
+
+    init(basicAuthorization: String? = nil, provider: @escaping CredentialsProvider) {
```

Comment:
Do we need to take auth in at init, should we prefer calling setBasicAuth at consumers who pass during init instead?

(this would centralise logic of encoding the values in one place)

I haven't seen all usage sites yet so maybe not appropriate!

Reply:
The `basicAuthorization` was not in use here. Removed.

Added the `AuthorizationStrategy` as an argument for `HttpAuthorizationProtocol`.